### DISCUSSION TITLE
Use wide class if no ToC

### DISCUSSION THREAD

**Discussion by tobiasvl** [mmistakes/minimal-mistakes#2496] [Q&A] [answered]
Is there a simple way to make the `single` layout always use `classes: wide`, _unless_ `toc: true` is also set?

In other words, I want posts to be wide by default so they fill the right side, unless I have specifically added a Table of Contents, in which case I want that to still appear on the right side.

Unless I'm mistaken, this is not easily achievable since you can't turn _off_ a class in front matter?

**Comment by mmistakes** [comment]
This is what Jekyll’s front matter defaults are for. Set the default to `classes: wide` and on those posts/pages where you set `toc: true` you can also set `classes: false` to disable it.

https://jekyllrb.com/docs/configuration/front-matter-defaults/

**Comment by iBug** [comment]
Might be easier to implement if you override the layout file to add the class automatically when `page.toc` is not enabled.

**Comment by tobiasvl** [comment]
@mmistakes Thanks, I knew about the front matter defaults but wasn't aware I could do `classes: false`

@iBug That sounds like a very good solution, thank you!

Edit: Or well, doesn't seem that easy since the class should be added to `<body>` in `default.html`. I guess I need to create a new layout.

**Comment by tobiasvl** [comment, marked as answer]
@iBug Thanks for getting me on the right track. There's probably a much better and shorter way to do it than this, but I'm not too familiar with Liquid. It works, though!

`_layouts/default.html`

```
  {% unless page.toc %}
    {% if page.classes or layout.classes %}
      {% assign classes = page.classes | default: layout.classes | push: 'wide' %}
    {% else %}
      {% assign classes = 'wide,' | split: ',' %}
    {% endif %}
  {% else %}
    {% if page.classes or layout.classes %}
      {% assign classes = page.classes | default: layout.classes %}
    {% else %}
      {% assign classes = '' | split: '' %}
    {% endif %}
  {% endunless %}

  <body class="layout--{{ page.layout | default: layout.layout }}{{ classes | join: ' ' | prepend: ' ' }}">
```
